### PR TITLE
Add Budget class for Buyer class

### PR DESCRIPTION
**Pull request by felixchanyy**
Addition of the Budget class is necessary to represent the budget attribute for Buyer objects.

Implemented the Budget class to encapsulate the budget information for buyers, allowing for better organization and management of budget-related data.

Using a separate class for budget ensures separation of concerns and follows the single responsibility principle, improving code readability and maintainability.

This commit also includes necessary adjustments to integrate the Budget class with the Buyer class.

### CODE REVIEW

**Review by lokidoki102 — APPROVED** [review]
Looks good to me

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 1]
```diff
@@ -16,4 +16,5 @@ public class CliSyntax {
     public static final Prefix PREFIX_UNITNUMBER = new Prefix("unitNo/");
     public static final Prefix PREFIX_BLOCK = new Prefix("blk/");
     public static final Prefix PREFIX_POSTALCODE = new Prefix("postal/");
+    public static final Prefix PREFIX_BUDGET = new Prefix("b/");
```
Comment (KhoonSun47) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:19]: Do you think it is better to explicitly name the prefix as budget/ ?

Not really a big issue, just a personal preference.

File: src/main/java/seedu/address/model/person/Budget.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        requireNonNull(budget);
+        checkArgument(isValidBudget(budget), MESSAGE_CONSTRAINTS);
+        value = budget;
+    }
+
+    /**
+     * Returns true if a given string is a valid budget amount.
+     */
+    public static boolean isValidBudget(String test) {
+        return test.matches(VALIDATION_REGEX) && Double.parseDouble(test) >= 0;
```
Comment (KhoonSun47) [inline src/main/java/seedu/address/model/person/Budget.java:31]: Can a budget be $0? And is there a minimum and maximum a budget can be?

Things to consider, depends on our team requirement (need to ask them).

**Review by KhoonSun47 — APPROVED** [review]
LGTM, just some comments to take note of.

File: src/main/java/seedu/address/logic/parser/CliSyntax.java (hunk 1 above)
Comment (felixchanyy) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java:19]: Should be fine I feel, like name is also "n/". We can discuss during meeting.

**Pull request merged**